pkg/addon/cmd.go

```diff
@@ -0,0 +1,88 @@
+package addon
+
+import (
+	"context"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	"github.com/spf13/cobra"
+	"github.com/stolostron/cluster-proxy-addon/pkg/constant"
+	"open-cluster-management.io/addon-framework/pkg/addonfactory"
+	"open-cluster-management.io/addon-framework/pkg/addonmanager"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func NewAddonCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "addon-manager",
+		Short: "addon-manager",
+		Long:  `The addon manager to deploy service-proxy and maintain certificates it using.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := runAddonManager(context.Background(), ctrl.GetConfigOrDie())
+			if err != nil {
+				klog.Fatal(err, "unable to run addon manager")
+			}
+		},
+	}
+	addFlags(cmd)
+	return cmd
+}
+
+var (
+	agentInstallAll       bool
+	certificatesNamespace string
+	agentImage            string
+)
+
+func addFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&certificatesNamespace, "certificates-namespace", "default", "The namespace where the secret is stored.")
+	cmd.Flags().BoolVar(&agentInstallAll, "agent-install-all", false, "Configure the install strategy of agent on managed clusters. Enabling this will automatically install agent on all managed cluster.")
+	cmd.Flags().StringVar(&agentImage, "agent-image", "", "The image of the agent.")
+}
+
+func runAddonManager(ctx context.Context, kubeConfig *rest.Config) error {
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	addMgr, err := addonmanager.New(kubeConfig)
+	if err != nil {
+		klog.Errorf("failed to new addon manager %v", err)
+		return err
+	}
+
+	registrationOption := NewRegistrationOption(
+		kubeConfig,
+		AddonName)
+
+	af := addonfactory.NewAgentAddonFactory(AddonName, FS, "manifests/charts/service-proxy").
+		WithGetValuesFuncs(GetValues(kubeClient, certificatesNamespace, agentImage), addonfactory.GetValuesFromAddonAnnotation).WithAgentRegistrationOption(registrationOption)
```

Comment:
TODO support addondeploymentconfig

Reply:
TODO added, will support `addondeploymentconfig` in other PR in the next Sprint.